### ISSUE TITLE
VULKAN setup is using CPU not GPU

### ISSUE THREAD

**Issue by stendavidson** [william-murray1204/stable-diffusion-cpp-python#7]
I successfully setup a VULKAN backend for stable-diffusion - at least in terms of installation everything went off without a hitch. However, as a test I run the sample code provided, however it defaults to using the CPU not the GPU:

![image](https://github.com/user-attachments/assets/43ada07c-a094-4da5-b42c-275ba4737bcf)

Here is the almost completely unmodified code sample:

![image](https://github.com/user-attachments/assets/3774fc05-36fb-4b23-80f9-e2dcf3fedf48)

**Comment by william-murray1204** [comment]
Hi @stendavidson, your code sample looks fine. 

What are the commands you are using to install `stable-diffusion-cpp-python`? Are you running Windows, Mac or Linux? 

On Windows 11 using the [VulkanSDK 1.3.296.0](https://vulkan.lunarg.com/sdk/home#windows), I am able to install SD cpp with a Vulkan backend by setting `DSD_VULKAN`  to `ON` :
```powershell
# Windows
$env:CMAKE_ARGS="-DSD_VULKAN=ON"
pip install stable-diffusion-cpp-python
```

If you are on Linux or Mac, you can install with this command instead:
```bash
# Linux and Mac
CMAKE_ARGS="-DSD_VULKAN=ON" pip install stable-diffusion-cpp-python
```

Output from running code sample:
```python
stable-diffusion.cpp:168  - Using Vulkan backend
ggml_vulkan: Found 2 Vulkan devices:
ggml_vulkan: 0 = AMD  Radeon RX 7800 XT (AMD proprietary driver) | uma: 0 | fp16: 1 | warp size: 64
ggml_vulkan: 1 = NVIDIA GeForce GTX 1050 Ti (NVIDIA) | uma: 0 | fp16: 0 | warp size: 32
ggml_vulkan: Compiling shaders..............................Done!
ggml_vulkan: Compiling shaders.............................Done!
...
```

If you are still encountering issues, please send me your GPU model, Vulkan version and any logs from the installation process. 

Another suggestion would be to try directly installing [stable-diffusion.cpp using Vulkan](https://github.com/leejet/stable-diffusion.cpp?tab=readme-ov-file#using-vulkan) to determine if the issue is coming from the Python bindings.

**Comment by stendavidson** [comment]
Hi @william-murray1204,

I am running on Windows however I have rather embarrassingly resolved the issue myself.

It appears that despite using the following command to reinstall with VULKAN (I attempted SYCL earlier and it worked but experienced some VRAM issues):

```
set CMAKE_ARGS=-DSD_VULKAN=ON
pip install stable-diffusion-cpp-python --upgrade --force-reinstall --no-cache-dir
```

pip didn't actually have the permissions to uninstall properly - I had to run in administrator mode to force:

`pip uninstall stable-diffusion-cpp-python`

Once that was complete the recommended setup succeeded.

Sorry for the hassle.
